torch/utils/data/dataloader.py

```diff
@@ -234,11 +262,23 @@ def _shutdown_workers(self):
         if not self.shutdown:
             self.shutdown = True
             self.done_event.set()
+            # if process_thread is waiting to put
+            if not self.data_queue.empty():
+                self.data_queue.get()
             for _ in self.workers:
                 self.index_queue.put(None)
+            # if all workers hang, still put None to let process_thread exit
+            if self.worker_result_queue.empty():
```

Comment:
As above. How would sending `None` to workers help with `process_thread`?

Reply:
This is meant to send the exit signal to the `process_thread`. The original code relies on workers to put `None` after receiving `None` so that `process_thread` can exit.